sites/cheerpj/src/content/docs/11-guides/basic-server-setup.md

```diff
@@ -0,0 +1,186 @@
+
+For properly serving compressed files without the Content-Encoding header, consider utilizing NGINX, which will be discussed further in the next section.
+
+## NGINX Guide
+
+We won't go into detail on how to install and setup NGINX, but will focus on the configuration necessary to integrate NGINX seamlessly with CheerpJ.
+
+For instructions on installing and setting up NGINX, you can refer to their official [Beginners Guide](https://nginx.org/en/docs/beginners_guide.html).
+Please note that the installation process may vary depending on your operating system. For guidance on installing NGINX on Windows, visit this [documentation](https://nginx.org/en/docs/windows.html) and for other operating systems, refer to [this](https://nginx.org/en/docs/install.html).
+
+### Nginx configuration file
+
+This NGINX configuration template will work with most Java applications and CheerpJ for testing purposes.
+
+```nginx
+worker_processes  1;
+error_log   nginx_main_error.log info;
+pid nginx_user.pid;
+daemon off;
+
+events {
+	worker_connections  1024;
+}
+
+http {
+	access_log  nginx_access.log;
+	error_log   nginx_error.log info;
+
+	include       /etc/nginx/mime.types;
+	default_type  application/octet-stream;
+
+	sendfile        on;
+
+	server {
+		listen       8080;
+		server_name  localhost;
+
+		gzip on;
+		gzip_types      application/javascript application/wasm text/plain;
+
+		charset utf-8;
+
+		location / {
+			# this should point to your root application folder
+			root /home/path/to/java/app;
+
+			autoindex on;
+			absolute_redirect off;
+			index  index.html index.htm;
+			add_header 'Cross-Origin-Opener-Policy' 'same-origin' always;
+			add_header 'Cross-Origin-Embedder-Policy' 'require-corp' always;
+			add_header 'Cross-Origin-Embedder-Policy' 'credentialless' always;
+		}
+	}
+}
+```
+
+Let's take a closer look at some important settings required for CheerpJ and to also ensure proper file serving.
+
+By including NGINX's `mime.types` file, NGINX effectively identifies the file type for each response and sets the corresponding `Content-Type` header in HTTP responses. This enables browsers and clients to interpret content accurately,
+
+```nginx
+include       /etc/nginx/mime.types;
+```
+
+Additionally, we specify the default MIME type for unspecified files as `application/octet-stream`. We do this to indicate that a file is binary data, instructing the server and client to treat the file as raw bytes instead of interpreting it as text or any other format.
+
+```nginx
+default_type  application/octet-stream;
+```
+
+We also enable compression for .wasm, .js, and .txt files, primarily used for the runtime chunks, within our server block.
+
+```nginx
+	gzip on;
+	gzip_types      application/javascript application/wasm text/plain;
```

Comment:
txt is not required. It does not hurt but it feels like it introduces more confusion than value.

.txt _used_ to be relevant in the CJ2 era